airflow/providers/amazon/aws/sensors/kinesis_analytics.py

```diff
@@ -0,0 +1,242 @@
+                    verify=self.verify,
+                    botocore_config=self.botocore_config,
+                ),
+                method_name="poke",
+            )
+        else:
+            super().execute(context=context)
+
+    def poke(self, context: Context, **kwargs) -> bool:
+        status = self.hook.conn.describe_application(ApplicationName=self.application_name)[
+            "ApplicationDetail"
+        ]["ApplicationStatus"]
+
+        self.log.info(
+            "Poking for AWS Managed Service for Apache Flink application: %s status: %s",
+            self.application_name,
+            status,
+        )
+
+        if status in self.FAILURE_STATES:
+            # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
+            if self.soft_fail:
+                raise AirflowSkipException(self.FAILURE_MESSAGE)
+            raise AirflowException(self.FAILURE_MESSAGE)
+
+        if status in self.SUCCESS_STATES:
+            self.log.info(
+                "AWS Managed Service for Apache Flink application started successfully `%s`.",
+                self.application_name,
+            )
+            return True
+
+        return False
+
+
+class KinesisAnalyticsV2StopApplicationCompletedSensor(AwsBaseSensor[KinesisAnalyticsV2Hook]):
+    """
+    Waits for AWS Managed Service for Apache Flink application to stop.
+
+    .. seealso::
+        For more information on how to use this sensor, take a look at the guide:
+        :ref:`howto/sensor:KinesisAnalyticsV2StopApplicationCompletedSensor`
+
+    :param application_name: Application name.
+
+    :param deferrable: If True, the sensor will operate in deferrable mode. This mode requires aiobotocore
+        module to be installed.
+        (default: False, but can be overridden in config file by setting default_deferrable to True)
+    :param poke_interval: Polling period in seconds to check for the status of the job. (default: 120)
+    :param max_retries: Number of times before returning the current state. (default: 75)
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+        If this is ``None`` or empty then the default boto3 behaviour is used. If
+        running Airflow in a distributed manner and aws_conn_id is None or
+        empty, then default boto3 configuration would be used (and must be
+        maintained on each worker node).
+    :param region_name: AWS region_name. If not specified then the default boto3 behaviour is used.
+    :param verify: Whether to verify SSL certificates. See:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html
+    :param botocore_config: Configuration dictionary (key-values) for botocore client. See:
+        https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html
+
+    """
+
+    aws_hook_class = KinesisAnalyticsV2Hook
+    ui_color = "#66c3ff"
+
+    INTERMEDIATE_STATES: tuple[str, ...] = (
```

Comment:
I think we usually prefer states to be in the hook as it allows users to be able also to utalize states for custom sensors/operators.

Reply:
@eladkal Sure thank you for the review is it fine in the next push i will make changes to move these states in to Hook?

Reply:
You can simply raise PR with this refactor. Since this code was not released it we can make changes

Reply:
Thank you @eladkal updated the changes. #40896